hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -301,9 +311,19 @@ public final UnaryOperator<T> projectRecord(Schema from, Schema to) {
    * @return the record position in the base file.
    */
   public long extractRecordPosition(T record, Schema schema, String fieldName, long providedPositionIfNeeded) {
+    if (supportsParquetRowIndex()) {
+      Object position = getValue(record, schema, fieldName);
+      if (position != null) {
+        return (long) position;
+      }
```

Comment:
Should we throw an error if the position is not read from the parquet reader when we enable that?

Reply:
sure